src/main/scala/com/sky/kafka/configurator/TopicConfigurationParser.scala

```diff
@@ -2,14 +2,42 @@ package com.sky.kafka.configurator
 
 import java.io.{Reader => JReader}
 
-import io.circe.yaml.parser.parse
+import cats.implicits._
+import io.circe
+import io.circe.generic.AutoDerivation
+import io.circe.yaml.parser._
+import io.circe.{Decoder, DecodingFailure, Json}
 
-import scala.util.Try
+object TopicConfigurationParser extends AutoDerivation {
 
-object TopicConfigurationParser {
-  def apply(topicConfigReader: JReader): Try[List[Topic]] =
+  def apply(topicConfigReader: JReader): Either[circe.Error, List[Topic]] =
     for {
-      ymlAsJson <- parse(topicConfigReader).toTry
-      topicConfigs <- ymlAsJson.as[List[Topic]].toTry
+      ymlAsJson <- parse(topicConfigReader)
+      topicConfigs <- ymlAsJson.as[List[Topic]]
     } yield topicConfigs
+
+  private case class TopicConfig(partitions: Int, replication: Int, config: Map[String, String])
```

Comment:
this and the decoders below were just moved over as is from the configurator package object